Assets/Scripts/GameManager.cs

```diff
@@ -81,13 +118,24 @@ public class GameManager : MonoBehaviour {
         gameOver = (players.Count == 1);
     }
 
-    void SetHealthTexts() {
-        if (player != null) {
-            playerHealthText.text = "Health: " + player.GetComponent<TankStats>().GetStat(StatType.MaxHealth);
-        }
-        if (computerPlayer != null) {
-            computerPlayerHealthText.text =
-                "Health: " + computerPlayer.GetComponent<TankStats>().GetStat(StatType.MaxHealth);
+    void SetStatsUIs() {
+        foreach (KeyValuePair<GameObject, Text> playerStatsUI in _playerStatsUis) {
+            if (playerStatsUI.Key != null) {
```

Comment:
When a player gets killed, the game object is destroyed. Resulting in:

![image](https://user-images.githubusercontent.com/6205060/28835177-d1b5acb4-76e4-11e7-9cd8-75565e9f4645.png)

The null check prevents the update for null objects